Filter settlements without user orders

vkgnosis opened the pull request:
Part of #1307

As an additional check to #1315 for all solvers. Even with this, 1315 still makes sense because we save time in the single order solvers by skipping liquidity orders.

### Test Plan
CI and new unit test

vkgnosis committed:
Filter settlements without user orders (#1316)

bh2smith reviewed (approved):
Noice.

fleupold reviewed (approved):
<a href="https://gitme.me/image?url=https%3A%2F%2Fmedia4.giphy.com%2Fmedia%2FPhKhSXofSAm3e%2Fgiphy.gif&token=noice" data-gitmeme-token="noice"><img src="https://media4.giphy.com/media/PhKhSXofSAm3e/giphy.gif" title="Created by gitme.me with /noice"/></a>

vkgnosis committed:
Do not settle liquidity orders in single order solver and baseline (#1315)

These solvers settle orders one by one without making use of liquidity
orders. Without this check they would treat them as user orders even
though we do not want these orders to be settled on their own.
